File: e2e-pytest-playwright/pages/login_page.py
from pages.abstract_page import AbstractPage

class LoginPage(AbstractPage):

  def enter_username(self, user):
    self.driver.fill("#username", user)
 
  def enter_password(self, password):
    self.driver.fill("#password", password)

  def click_sign_in(self):
    self.driver.click("input:has-text('Sign in')")
    

File: e2e-junit-selenium/src/pages/LoginPage.java
package pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

public class LoginPage < AbstractPage {

    public LoginPage(WebDriver aDriver) {
        super(aDriver);
    }

    public void enterUsername(String username) {
        driver.findElement(By.id("username")).clear();
        driver.findElement(By.id("username")).sendKeys(username);
    }

    public void enterPassword(String password) {
        driver.findElement(By.id("password")).clear();
        driver.findElement(By.id("password")).sendKeys(password);
    }

    public void clickSignIn() {
        driver.findElement(By.xpath("//input[@value='Sign in']")).click();
    }
}
